### PR TITLE
Support for changes in ConnectionMonitor API

### PR DESCRIPTION
**Pull request by anton-evseev**
ConnectionMonitor won't be supported beyond API version 2019-06-01 (there will be a V2 version added later), this PR is to address that and avoid introducing breaking changes down the line. What it does:
- Separates ConnectionMonitor specs from NewtorkWatcher into it's own JSON file
- The package for 2019-06-01 will be basically the same
- The package for 2019-07-01 will reference ConnectionMonitor specs from 2019-06-01 so the future SDK packages won't break

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by number213** [comment]
SDK java failed with timeout (as usual)...

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/networkWatcherConnectionMonitorV1.json [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkWatchers/{networkWatcherName}/connectionMonitors/{connectionMonitorName}": {
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/networkWatcherConnectionMonitorV1.json]: ConnectionMonitor is a tracked resource, and requires a PATCH api to update tags. Standard clients assume all tracked resources support a PATCH api for tags

**Review by KrisBash — CHANGES_REQUESTED** [review]
Changes look good to me, but a PATCH api is needed for connectionMonitor to support tags update

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/networkWatcherConnectionMonitorV1.json (hunk 1 above)
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/networkWatcherConnectionMonitorV1.json]: Added PATCH for ConnectionMonitor

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by number213** [comment]
Rebased PR to include fixes for CI that were merged into `network-august-branch` recently

**Pull request merged**